pkg/verify/verify.go

```diff
@@ -0,0 +1,469 @@
+package verify
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/transport"
+
+	"github.com/golang/glog"
+	"golang.org/x/crypto/openpgp"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/cluster-version-operator/pkg/payload"
+)
+
+// Interface performs verification of the provided content. The default implementation
+// in this package uses the container signature format defined at https://github.com/containers/image
+// to authenticate that a given release image digest has been signed by a trusted party.
+type Interface interface {
+	Verify(ctx context.Context, releaseDigest string) error
+}
+
+type rejectVerifier struct{}
+
+func (rejectVerifier) Verify(ctx context.Context, releaseDigest string) error {
+	return fmt.Errorf("verification is not possible")
+}
+
+// Reject fails always fails verification.
+var Reject Interface = rejectVerifier{}
+
+// LoadFromPayload looks for a config map in the v1 API group within within the provided update
+// with the annotation "release.openshift.io/verification-config-map". Only the first payload
+// entry with the annotation is used.
+//
+// The presence of this config map instructs the CVO to verify updates before they are downloaded.
+//
+// The keys within the config map define how verification is performed:
+//
+// verifier-public-key-*: One or more GPG public keys in ASCII form that must have signed the
+//                        release image by digest.
+//
+// store-*: A URL (scheme file://, http://, or https://) location that contains signatures. These
```

Comment:
nit: we should not allow file:// right? there is not way we can make that happen...

Reply:
I'll update the note - I allow it for testing both paths and it's possible we could mount it in to the CVO in the future.  Testing only for now.